hammerblade/torch/kernel/kernel_softmax_backward.cpp

```diff
@@ -0,0 +1,63 @@
+#include <kernel_common.hpp>
+#include <stdint.h>
+#include <cmath>
+
+// We wrap all external-facing C++ kernels with `extern "C"` to
+// prevent name mangling
+
+extern "C" {
+
+  //====================================================================
+  // LogSoftMax Backward kernel
+  // 03/26/2020 Lin Cheng
+  //====================================================================
+
+  __attribute__ ((noinline))  int tensorlib_log_softmax_backward(
+          bsg_tensor_t* grad_input_p,
+          bsg_tensor_t* grad_p,
+          bsg_tensor_t* output_p,
+          int32_t*      dim_p) {
+
+    BSGTensor<float> gI(grad_input_p);
+    BSGTensor<float> grad(grad_p);
+    BSGTensor<float> output(output_p);
```

Comment:
Guess these are unused?

Reply:
I think this is true for `gI` and `output`